src/commands/create.rs

```diff
@@ -0,0 +1,233 @@
+              "name": "test-identity",
+              "organizationId": "org-xyz-456",
+              "clientSecret": "ssh-rsa",
+              "credentialExpiration": "2025-12-31T23:59:59Z"
+            }
+          }
+        }
+        "#;
+
+        let mock_http_client = setup_mock_http_client(full_expected_response);
+        let iam_client = HttpIAMClient::new(mock_http_client, metadata.clone());
+        let command = CreateCommand::new(iam_client);
+        let token = dummy_token();
+        let result = command.create_l1_system_identity(token);
+        assert!(
+            result.is_ok(),
+            "L1 creation failed when it should have succeeded. Error: {:?}",
+            result.err()
+        );
+
+        assert_eq!(result.unwrap(), expected_identity);
+    }
+    #[test]
+    fn test_create_l2_system_identity_success() {
+        let metadata = create_test_metadata(Some("/tmp"), Some("l2_identity_test".to_string()));
+        let expected_identity = SystemIdentity {
+            id: "identity-123".to_string(),
+            name: Some("test-identity".to_string()),
+            client_id: "client-abc-789".to_string(),
+            organization_id: "org-xyz-456".to_string(),
+            identity_type: SystemIdentityType::L2 {
+                pub_key: "cHVibGljS2V5QmFzZTY0RW5jb2RlZFN0cmluZw==".to_string(),
+            },
+        };
+
+        let full_expected_response = r#"
+    {
+      "data": {
+        "systemIdentityCreate": {
+          "clientId": "client-abc-789",
+          "publicKey": "cHVibGljS2V5QmFzZTY0RW5jb2RlZFN0cmluZw==",
+          "id": "identity-123",
+          "name": "test-identity",
+          "organizationId": "org-xyz-456"
+        }
+      }
+    }
+    "#;
+
+        let mock_http_client = setup_mock_http_client(full_expected_response);
+        let iam_client = HttpIAMClient::new(mock_http_client, metadata.clone());
+        let command = CreateCommand::new(iam_client);
+        let token = dummy_token();
+
+        let result = command.create_l2_system_identity(&metadata, token);
+        assert!(
+            result.is_ok(),
+            "L2 creation failed when it should have succeeded. Error: {:?}",
+            result.err()
+        );
+
+        assert_eq!(result.unwrap(), expected_identity);
+    }
+    #[test]
+    fn test_create_l2_system_identity_malformed_response() {
+        let metadata = create_test_metadata(Some("/tmp"), Some("l2_identity_test".to_string()));
+        let malformed_response = "{ invalid json }";
+
+        let mock_http_client = setup_mock_http_client(malformed_response);
+        let iam_client = HttpIAMClient::new(mock_http_client, metadata.clone());
+        let command = CreateCommand::new(iam_client);
+        let token = dummy_token();
+        let result = command.create_l2_system_identity(&metadata, token);
+
+        assert!(
+            result.is_err(),
+            "Expected deserialization error but got success"
+        );
```

Comment:
Nit:
```suggestion
        assert_matches!(
            result,
            Err(CreateError(_)),
        );
```

Same in test below